### PR TITLE
Issue 67: do not shut down pumps so often

### PR DESCRIPTION
**Pull request by JamesBirdsall**
## Description

Equivalent change to recent dotnet fix. Only shut down a pump on ReceiverDisconnectedException.
Related update to README to better describe threading model.
One-line test fix.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [ ] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [ ] If applicable, the public code is properly documented.
- [ ] Pull request includes test coverage for the included changes.
- [ ] The code builds without any errors.

### CODE REVIEW

File: azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/EventHubPartitionPump.java [hunk 1]
```diff
@@ -211,7 +211,6 @@ public void onReceive(Iterable<EventData> events)
 		@Override
 		public void onError(Throwable error)
```
Comment (serkantkaraca) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/EventHubPartitionPump.java]: >onError [](start = 14, length = 7)

ReceivePump behavior should also change. ReceivePump shuts down after delivering the error to the handler. Or maybe, PartitionManager can reset the handler on error.

Comment (JamesBirdsall) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/EventHubPartitionPump.java]: If the pump calls the error handler, it shuts down, but I'm not sure what it takes to get the pump to declare an error.  While trying to test the changes I made to address this, I experimented with forcibly closing the TCP connection and the client recovered at a lower level. The pump never saw the error.

Comment (serkantkaraca) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/EventHubPartitionPump.java]: Client retry probably re-established the connection in your experiment and that is how it recovered. Try disabling the retry, that would stop the pump.

**Comment by JamesBirdsall** [comment]
The problem that Java is experiencing is not the same as Dotnet and this is the wrong approach. Closing this PR.

**Pull request closed without merging**